nextcord/http.py

```diff
@@ -240,151 +705,326 @@ async def request(
+                    continue
+
+                raise
+
+            except RateLimitMigrating as e:
+                if url_rate_limit.migrating is None:
+                    raise ValueError(
+                        "RateLimitMigrating raised, but RateLimit.migrating is None. This is an internal Nextcord "
+                        "error and should be reported!"
+                    ) from e
+
+                old_rate_limit = url_rate_limit
+                url_rate_limit = self._buckets.get(url_rate_limit.migrating)
+                if url_rate_limit is None:
+                    # This means we have an internal issue that we need to fix.
+                    _log.error(
+                        "RateLimit said to migrate, but the RateLimit to migrate to was not found. This is an "
+                        "internal Nextcord error and should be reported.\n"
+                        "Migrating RateLimit.migrate_to: %s\n"
+                        "Migrating RateLimit.bucket: %s\n"
+                        "Route: %s",
+                        old_rate_limit.migrating,
+                        old_rate_limit.bucket,
+                        route,
+                        exc_info=e,
+                    )
+                    raise ValueError(
+                        "RateLimit said to migrate, but the RateLimit to migrate was not found? This is an "
+                        "internal Nextcord error and should be reported!"
+                    ) from e
+
+            else:
+                if not should_retry:
+                    break
+
+            if retry_count >= max_retry_count - 1:
+                _log.error(
+                    "Hit retry %s/%s on %s, either something is wrong with Discord or Nextcord.",
+                    retry_count + 1,
+                    max_retry_count,
+                    rate_limit_path,
+                )
+                if response is not None:
+                    if response.status >= 500:
+                        raise DiscordServerError(response, ret)
 
-            if response is not None:
-                # We've run out of retries, raise.
-                if response.status >= 500:
-                    raise DiscordServerError(response, data)
+                    raise HTTPException(response, ret)
 
-                raise HTTPException(response, data)
+        return ret
+
+    async def _handle_http_response_errors(
+        self,
+        response: aiohttp.ClientResponse,
+        return_value: Any,
+        rate_limit_path: Tuple[str, str, Optional[str]],
+        retry_request: bool,
+        retry_count: int,
+        url_rate_limit: RateLimit,
+        global_rate_limit: GlobalRateLimit,
+    ) -> bool:
+        """|coro|
+        This exists to make the main request function smaller and easier to work with.
+
+        I'm on a 1440p 21:9 monitor and even with PyCharm split into multiple panes, the request method with all of
+        this in it is absolutely huge.
```

Comment:
I think this would be better off as a comment above the function instead of in the documentation.
```suggestion

```